src/test/java/net/snowflake/client/jdbc/RestRequestWiremockTest.java

```diff
@@ -0,0 +1,250 @@
+    private static final String WIREMOCK_HOME_DIR = ".wiremock";
+    private static final String WIREMOCK_M2_PATH =
+            "/.m2/repository/org/wiremock/wiremock-standalone/3.8.0/wiremock-standalone-3.8.0.jar";
+    private static final String WIREMOCK_HOST = "localhost";
+    private static final String TRUST_STORE_PROPERTY = "javax.net.ssl.trustStore";
+    private static int httpProxyPort;
+    private static int httpsProxyPort;
+    private static String originalTrustStorePath;
+    private static Process wiremockStandalone;
+
+    private static void startWiremockStandAlone() {
+        // retrying in case of fail in port bindings
+        await()
+                .alias("wait for wiremock responding")
+                .atMost(Duration.ofSeconds(10))
+                .until(
+                        () -> {
+                            try {
+                                httpProxyPort = findFreePort();
+                                httpsProxyPort = findFreePort();
+                                wiremockStandalone =
+                                        new ProcessBuilder(
+                                                "java",
+                                                "-jar",
+                                                getWiremockStandAlonePath(),
+                                                "--root-dir",
+                                                System.getProperty("user.dir")
+                                                        + File.separator
+                                                        + WIREMOCK_HOME_DIR
+                                                        + File.separator,
+                                                "--enable-browser-proxying", // work as forward proxy
+                                                "--proxy-pass-through",
+                                                "false", // pass through only matched requests
+                                                "--port",
+                                                String.valueOf(httpProxyPort),
+                                                "--https-port",
+                                                String.valueOf(httpsProxyPort),
+                                                "--https-keystore",
+                                                getResourceURL("wiremock" + File.separator + "ca-cert.jks"),
+                                                "--ca-keystore",
+                                                getResourceURL("wiremock" + File.separator + "ca-cert.jks"))
+                                                .inheritIO()
+                                                .start();
+                                waitForWiremock();
+                                return true;
+                            } catch (Exception e) {
+                                logger.warn("Failed to start wiremock, retrying: ", e);
```

Comment:
I'm wondering maybe that one day for some reason it may hang forever. So maybe a retry max count would be nice to have and when reached this exception rethrown, makes sense?